### DISCUSSION TITLE
How to use Debouncer with hxAuttoSuggest and Search?

### DISCUSSION THREAD

**Discussion by chobo2** [havit/Havit.Blazor#694] [Q&A]
Hi

I am confused on how to use the debouncer with AutoSuggest and Search based on the debouncer example

https://havit.blazor.eu/concepts/Debouncer

I first thought I would put the debouncer in 
```

	private async Task<AutosuggestDataProviderResult<EmployeeDto>> ProvideSuggestions(AutosuggestDataProviderRequest request)
	{
		var matchingEmployees = await DemoDataService.FindEmployeesByNameAsync(request.UserInput, limitCount: 10, request.CancellationToken);

		return new AutosuggestDataProviderResult<EmployeeDto> { Data = matchingEmployees };
	}

```

thought I would do like this

```

	private async Task<AutosuggestDataProviderResult<EmployeeDto>> ProvideSuggestions(AutosuggestDataProviderRequest request)
	{
          await queryDebouncer.DebounceAsync(millisecondsDelay: 500, actionAsync: async (cancellationToken) =>
		          {
			        	var matchingEmployees = await DemoDataService.FindEmployeesByNameAsync(request.UserInput, limitCount: 10, request.CancellationToken);

		return new AutosuggestDataProviderResult<EmployeeDto> { Data = matchingEmployees };
		          });
	}

```

but it won't allow me to return it. So how do I do this as I thought I always need to return an AutosuggestDAtaProviderResult

**Comment by hakenr** [comment]
`HxAutosuggest` does have built-in debouncing, just set the `Delay` parameter (default is 300ms).

**Reply by chobo2** [reply]
Okay yeah I saw that but was not sure if it was the same thing or differenta as I saw this

> Consider caching the initial suggestions to avoid unnecessary roundtrips for repetitive DataProvider calls.
> You might want to tweak the Delay parameter to disable/minimize the delay before the initial suggestions are displayed.
> **In such case, consider using additional [Debouncer](https://havit.blazor.eu/concepts/Debouncer) for regular suggestions to avoid unnecessary roundtrips on every keystroke.**

Then not sure why I am getting a problem of sometimes the auto suggestions just don't show up. it seems like it tries to open up but then it does not. I have to click into the input box and then that triggers the auto suggestions to show again.
